`ci / hive-tests (config-hdp3)`, `pt (default, suite-gcs, )` fail due to GIB when only Iceberg tests modified

https://github.com/trinodb/trino/actions/runs/7489779443/job/20387125843?pr=20328

```
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  2.[71](https://github.com/trinodb/trino/actions/runs/7489779443/job/20387125843?pr=20328#step:5:72)7 s
[INFO] Finished at: 2024-01-11T14:23:48Z
[INFO] ------------------------------------------------------------------------
Error:  Failed to execute goal on project trino-hive: Could not resolve dependencies for project io.trino:trino-hive:trino-plugin:436-SNAPSHOT: The following artifacts could not be resolved: io.trino:trino-cache:jar:436-SNAPSHOT (absent), io.trino:trino-filesystem:jar:436-SNAPSHOT (absent), io.trino:trino-filesystem-manager:jar:436-SNAPSHOT (absent), io.trino:trino-hive-formats:jar:436-SNAPSHOT (absent), io.trino:trino-memory-context:jar:436-SNAPSHOT (absent), io.trino:trino-orc:jar:436-SNAPSHOT (absent), io.trino:trino-parquet:jar:436-SNAPSHOT (absent), io.trino:trino-plugin-toolkit:jar:436-SNAPSHOT (absent), io.trino:trino-spi:jar:436-SNAPSHOT (absent), io.trino:trino-filesystem-s3:jar:436-SNAPSHOT (absent): Cannot access sonatype-nexus-snapshots (https://oss.sonatype.org/content/repositories/snapshots) in offline mode and the artifact io.trino:trino-cache:jar:436-SNAPSHOT has not been downloaded from it before. -> [Help 1]
Error:  
Error:  To see the full stack trace of the errors, re-run Maven with the -e switch.
Error:  Re-run Maven using the -X switch to enable full debug logging.
Error:  
Error:  For more information about the errors and possible solutions, please read the following articles:
Error:  [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/DependencyResolutionException
Error: Process completed with exit code 1.
```

apparently `pt (default, suite-gcs, )` too:

```
tests               | 2024-01-11 20:41:23 INFO: FAILURE     /    io.trino.tests.product.TestConfiguredFeatures.selectConfiguredConnectors (Groups: configured_features) took 0.3 seconds
tests               | 2024-01-11 20:41:23 SEVERE: Failure cause:
tests               | java.lang.AssertionError: Could not find rows:
tests               | [delta]
tests               | 
tests               | actual rows:
tests               | [hive]
tests               | [tpch]
tests               | [delta_lake]
tests               | [iceberg]
tests               | [blackhole]
tests               | [system]
tests               | [jmx]
```

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -310,7 +310,7 @@ jobs:
           S3_BUCKET: ${{ vars.TRINO_S3_BUCKET }}
           S3_BUCKET_ENDPOINT: "s3.${{ vars.TRINO_AWS_REGION }}.amazonaws.com"
         run: |
-          if [ "${AWS_ACCESS_KEY_ID}" != "" ]; then
+          if [ "${AWS_ACCESS_KEY_ID}" != "" ] && ( [ ! -f gib-impacted.log ] || grep -q plugin/trino-hive gib-impacted.log ); then
             $MAVEN test ${MAVEN_TEST} -pl :trino-hive -P aws-tests
           fi
       - name: Upload test results

